sys/include/memarray.h

```diff
@@ -29,19 +29,27 @@ extern "C" {
  * @brief Memory pool
  */
 typedef struct {
-    void *first_free; /**< head of the free list */
+    void *first_free;   /**< head of the free list */
+    size_t size;        /**< size of single list element */
+    size_t num;         /**< max number of elements in list */
 } memarray_t;
 
 /**
  * @brief Define static memory pool
  *
- * @param[out] name    name for defined memarray pool
- * @param[in]  size    size of single memory chunk
- * @param[in]  num     number of chunks in pool
+ * @param[out] name       name for defined memarray pool
+ * @param[in]  structure  structure in single memory chunk
+ * @param[in]  numb       number of chunks in pool
  */
-#define MEMARRAY(name, size, num)                                         \
-    static uint8_t _data_ ## name[num * size];                            \
-    static memarray_t name = { .first_free = _data_ ## name };
+#define MEMARRAY(name, structure, numb)                                           \
+    union el {                                                                    \
```

Comment:
One issue with the declaration of `union el` here is that you now cannot invoke `MEMARRAY` twice:

```shell
n file included from /home/bergmann/src/RIOT/examples/dtls-echo/bin/pkg/native/tinydtls/crypto.c:122:0:
/home/bergmann/src/RIOT/sys/include/memarray.h:45:11: error: redefinition of ‘union el’
     union el {                                                                    \
           ^
/home/bergmann/src/RIOT/examples/dtls-echo/bin/pkg/native/tinydtls/crypto.c:124:1: note: in expansion of macro ‘MEMARRAY’
 MEMARRAY(security_storage,  dtls_security_parameters_t, DTLS_SECURITY_MAX)
 ^~~~~~~~
/home/bergmann/src/RIOT/sys/include/memarray.h:45:11: note: originally defined here
     union el {                                                                    \
           ^
/home/bergmann/src/RIOT/examples/dtls-echo/bin/pkg/native/tinydtls/crypto.c:123:1: note: in expansion of macro ‘MEMARRAY’
 MEMARRAY(handshake_storage, dtls_handshake_parameters_t, DTLS_HANDSHAKE_MAX)
 ^~~~~~~~
```

Reply:
@obgm: absolutely right, thanks for noticing. I'm gonna append `name` as i did with `_data ## name`